src/app/core/store/courses/courses.effects.ts

```diff
@@ -0,0 +1,126 @@
+import { Injectable } from "@angular/core";
+import { Actions, createEffect, ofType } from "@ngrx/effects";
+import { Action } from "@ngrx/store";
+import { Observable, of as observableOf, EMPTY } from "rxjs";
+import { switchMap, map, catchError, mergeMap, take } from "rxjs/operators";
+import sortBy from "lodash/sortBy";
+import isNil from "lodash/isNil";
+import last from "lodash/last";
+
+import * as CoursesActions from "./courses.actions";
+import { CoursesListService, CourseItemModel } from "../../services/index";
+import assign from "lodash/assign";
+
+@Injectable()
+export class CoursesEffects {
+  private actions$: Actions;
+  private coursesListService: CoursesListService;
+
+  /**
+   * Effect for processing getCourses
+   */
+  public getCourses$: Observable<Action>;
+
+  /**
+   * Effect for processing updateCourse
+   */
+  public updateCourse$: Observable<Action>;
+
+  /**
+   * Effect for processing deleteCourse
+   */
+  public deleteCourse$: Observable<Action>;
+
+  /**
+   * Effect for processing createCourse
+   */
+  public createCourse$: Observable<Action>;
+
+  /**
+   * Effect for processing searchCourses
+   */
+  public searchCourses$: Observable<Action>;
+
+  /**
+   * Effect for processing loadMoreCourses
+   */
+  public loadMoreCourses$: Observable<never>;
+
+  constructor(actions$: Actions, coursesListService: CoursesListService) {
+    this.actions$ = actions$;
+    this.coursesListService = coursesListService;
+
+    this.getCourses$ = createEffect(() =>
+      this.actions$.pipe(
+        ofType(CoursesActions.getCourses),
+        switchMap(action => this.coursesListService.getCoursesList(action.isLimited)),
```

Comment:
this.coursesListService.getCoursesList(action.isLimited).pipe( and move here map and catchError)